### PR TITLE
Allow per-request tags for statsd

### PR DESCRIPTION
**Pull request by urmastalimaa**
Helps to differentiate kinds of requests in metrics (reporting,
read-only, maintenance, etc).

### CODE REVIEW

File: lib/salestation/web/statsd_middleware.rb [hunk 1]
```diff
@@ -23,11 +24,13 @@ def call(env)
...
 
-        @statsd.timing(@metric, (Time.now - start) * 1000, tags: [
+        tags = [
           "path:#{ path }",
           "method:#{ method }",
           "status:#{ status }"
-        ])
+        ] + env.fetch(EXTRA_TAGS_ENV_KEY, [])
+
+        @statsd.timing(@metric, (Time.now - start) * 1000, tags: tags)
```
Comment (take-five) [inline lib/salestation/web/statsd_middleware.rb]: Hmm, I though I fixed how the request duration is measured.

Comment (take-five) [reply lib/salestation/web/statsd_middleware.rb]: Ah, that was here https://github.com/salemove/salestation/blob/master/lib/salestation/web/request_logger.rb

I know it's out of scope, but perhaps you could fix it here too?

File: salestation.gemspec [hunk 2]
```diff
@@ -4,7 +4,7 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 
 Gem::Specification.new do |spec|
   spec.name          = "salestation"
-  spec.version       = "3.4.0"
+  spec.version       = "3.4.1"
```
Comment (indrekj) [inline salestation.gemspec]: bump minor version for features

File: lib/salestation/web/statsd_middleware.rb (hunk 1 above)
Comment (urmastalimaa) [reply lib/salestation/web/statsd_middleware.rb]: Can you elaborate (maybe that explains some of the bad latency metrics?)

Comment (take-five) [reply lib/salestation/web/statsd_middleware.rb]: See this commit: https://github.com/salemove/salestation/commit/e9fef41f7447edda566b715ff03c46eced7a321d

File: salestation.gemspec (hunk 2 above)
Comment (urmastalimaa) [reply salestation.gemspec]: oopsie

File: lib/salestation/web/statsd_middleware.rb (hunk 1 above)
Comment (urmastalimaa) [reply lib/salestation/web/statsd_middleware.rb]: https://github.com/salemove/salestation/pull/39/commits/8510b63f37639c7bb0c2795cb911378416cf16f2

**Comment by urmastalimaa** [comment]
!squash

**Pull request merged**